core/src/telemetry/otel.rs

```diff
@@ -320,24 +325,25 @@ fn default_resource(override_values: &HashMap<String, String>) -> Resource {
     let override_kvs = override_values
         .iter()
         .map(|(k, v)| KeyValue::new(k.clone(), v.clone()));
-    default_resource_instance()
-        .clone()
-        .merge(&Resource::new(override_kvs))
-}
-
-#[derive(Clone)]
-struct ConstantTemporality(Temporality);
-
-impl TemporalitySelector for ConstantTemporality {
-    fn temporality(&self, _: InstrumentKind) -> Temporality {
-        self.0
-    }
+    let override_resource = Resource::builder().with_attributes(override_kvs).build();
+    Resource::builder_empty()
+        .with_attributes(
+            default_resource_instance()
+                .iter()
+                .map(|(k, v)| KeyValue::new(k.clone(), v.clone())),
+        )
+        .with_attributes(
+            override_resource
+                .iter()
+                .map(|(k, v)| KeyValue::new(k.clone(), v.clone())),
+        )
+        .build()
```

Comment:
https://github.com/open-telemetry/opentelemetry-rust/pull/2322 👀 

the removal of .merge is a bit of a bummer. 

see https://github.com/open-telemetry/opentelemetry-rust/issues/2766